modules/admin/admin-ui/src/main/resources/web/admin/common/js/liveedit/PageView.ts

```diff
@@ -197,7 +203,23 @@ module api.liveedit {
                 this.resetAction.setEnabled(false);
             }
 
-            this.addContextMenuActions([inspectAction, this.resetAction]);
+            this.saveAsTemplateAction = new api.ui.Action('Save as Template').onExecuted(action => {
+                let content = this.liveEditModel.getContent();
+                new CreatePageTemplateRequest()
+                    .setController(this.pageModel.getController().getKey())
+                    .setRegions(this.pageModel.getRegions())
+                    .setConfig(this.pageModel.getConfig())
+                    .setDisplayName(content.getDisplayName())
+                    .setSite(content.getPath())
+                    .setSupports(content.getType())
+                    .setName(content.getName())
+                    .sendAndParse().then(createdTemplate => {
+
+                    new EditContentEvent([ContentSummaryAndCompareStatus.fromContentSummary(createdTemplate)]).fire();
+                });
+            });
+
```

Comment:
I suggest extracting SaveAsTemplateAction into a separate class which both PageView and PageInspectionPanel classes will use. As of now this code is duplicated in these two classes.